frontend/src/modals/new-object/NewObjectModal.tsx

```diff
@@ -40,99 +65,165 @@ const createChildMutation = gql`
-      return newState;
-    });
+  const onChildCreationDescriptionChange = (event) => {
+    const value = event.target.value;
+    const changeChildCreationDescriptionEvent: ChangeChildCreationDescriptionEvent = {
+      type: 'CHANGE_CHILD_CREATION_DESCRIPTION',
+      childCreationDescriptionId: value,
+    };
+    dispatch(changeChildCreationDescriptionEvent);
   };
 
-  // Create the new child
-  const [createChild, { loading, data, error }] = useMutation(createChildMutation);
-  const onCreateChild = (event) => {
-    event.preventDefault();
+  const [createChild, { loading: createChildLoading, data: createChildData, error: createChildError }] = useMutation<
+    GQLCreateChildMutationData
+  >(createChildMutation);
+  useEffect(() => {
+    if (!createChildLoading) {
+      if (createChildError) {
+        const showToastEvent: ShowToastEvent = {
+          type: 'SHOW_TOAST',
+          message: 'An unexpected error has occurred, please refresh the page',
+        };
+        dispatch(showToastEvent);
+      }
+      if (createChildData) {
+        const handleResponseEvent: HandleResponseEvent = { type: 'HANDLE_RESPONSE', data: createChildData };
+        dispatch(handleResponseEvent);
+      }
+    }
+  }, [createChildLoading, createChildData, createChildError, dispatch]);
+
+  const onCreateObject = () => {
+    dispatch({ type: 'CREATE_CHILD' } as CreateChildEvent);
     const input = {
       id: uuid(),
-      projectId,
+      editingContextId,
       objectId,
       childCreationDescriptionId: selectedChildCreationDescriptionId,
     };
     createChild({ variables: { input } });
   };
+
   useEffect(() => {
-    if (!loading && !error && data?.createChild?.object) {
-      onObjectCreated(data.createChild.object);
+    if (newObjectModal === 'success') {
+      onObjectCreated(objectToSelect);
     }
-  }, [loading, data, error, onObjectCreated]);
+  }, [newObjectModal, onObjectCreated, objectToSelect]);
 
   return (
-    <Modal title="Create a new object" onClose={onClose}>
-      <Form onSubmit={onCreateChild}>
-        <Label value="Object type">
-          <Select
-            name="stereotype"
-            value={selectedChildCreationDescriptionId}
-            options={childCreationDescriptions}
-            onChange={setSelectedChildCreationDescriptionId}
-            autoFocus
-            data-testid="stereotype"
-          />
-        </Label>
-        <Buttons>
-          <ActionButton type="submit" label="Create" data-testid="create-object" />
-        </Buttons>
-      </Form>
-    </Modal>
+    <>
+      <Dialog open={true} onClose={onClose} aria-labelledby="dialog-title" maxWidth="xs" fullWidth>
+        <DialogTitle id="dialog-title">Create a new object</DialogTitle>
+        <DialogContent>
+          <div className={classes.form}>
+            <InputLabel id="newObjectModalChildCreationDescriptionLabel">Object type</InputLabel>
+            <Select
+              value={selectedChildCreationDescriptionId}
```

Comment:
I get warnings from MUI when using this and the initial state has the `selectedChildCreationDescriptionId` to `null`. I tried ` || ''` (based on the message from MUI) and it seems to work but I'm not sure if this is the best way.

Reply:
I'll update the state machine for that